ipamutils/utils.go

```diff
@@ -48,3 +60,75 @@ func initGranularPredefinedNetworks() []*net.IPNet {
 	}
 	return pl
 }
+
+// InitAddressPools allows to initialize the local and global scope predefined
+// address pools to the desired values. It fails is invalid input is passed
+// or if the predefined pools were already initialized.
+func InitAddressPools(list []*PredefinedPools) error {
+	var (
+		localPools []*net.IPNet
+		done       bool
+	)
+
+	for _, p := range list {
+		if p == nil {
+			continue
+		}
+		_, b, err := net.ParseCIDR(p.Base)
+		if err != nil {
+			return fmt.Errorf("invalid base pool %q: %v", p.Base, err)
+		}
+		ones, _ := b.Mask.Size()
+		if p.Size <= 0 || p.Size < ones {
+			return fmt.Errorf("invalid pools size: %d", p.Size)
+		}
+		localPools = append(localPools, initPools(p.Size, b)...)
+
+	}
+
+	if localPools == nil {
+		localPools = initBroadPredefinedNetworks()
+	}
+
+	initNetworksOnce.Do(func() {
+		PredefinedBroadNetworks = localPools
+		done = true
+	})
+
+	if !done {
+		return fmt.Errorf("predefined address pools are already initialized")
```

Comment:
The point of using the sync.Once construct is to avoid double init, is there a reason for which we have to fail here?